src/app/navbar/navbar.component.ts

```diff
@@ -47,4 +56,24 @@ export class NavbarComponent implements OnInit {
       this.isCollapsed = true;
     }
 
-}
+    getTemperature(callback): void{
+      var request = new XMLHttpRequest();
+      var data;
+      var temperature;
+      var icon;
+
+      request.open('GET', 'http://api.openweathermap.org/data/2.5/weather?units=imperial&zip=90024,us&APPID=bc6a73dfabbd4e6c9006a835d00589f2', true);
```

Comment:
Rather than have this really long request url, break it up into variables

e.g. API_KEY, baseWeatherUrl, zipcode
then you can form the query like "let weatherQuery = `${baseWeatherUrl}?units=imperial&zip=${zipcode},us&APPID=${apiKey}`;"